### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by kathrynljackson**
### Participating Group Members:
* @kathrynljackson 
* @nathanielmillard 
* @stacyp2006 
* @aemcdonald 
 
### Code Highlights:
 * Create Home, Profile, and Subscriptions test files
 * Add testing libraries
 * Begin testing Profile to ensure rendering
* Passed Tests: 
   * should set user state
   * show status of login request
   * login response status
   * logout status
   * return initial state

### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing. 
- [x] Yes, and all are passing.
 
### Any background context you want to provide?
We used the following learning resources:

- [Testing React](https://www.youtube.com/watch?v=AMo3d_K6e2A)
- [Testing Redux](https://www.youtube.com/watch?v=vbvQzWDCuXU)
- [A nice way to test redux reducers and actions](https://blog.thepete.net/blog/2019/07/15/a-nice-way-to-test-redux-reducers-and-actions---part-i/)
- [.objectContaining](https://stackoverflow.com/questions/50950704/expect-objectcontaining-is-not-a-function-in-jest/50950853)

### Issues: 
* Addresses #22 
* Addresses #21  
* Addresses #24 
 
### Screenshots (if appropriate): 
n/a
 
### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

File: code/web/src/modules/user/api/state.test.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            type: LOGOUT
+        }
+
+        const initialState = {
+            error: null,
+            isLoading: false,
+            isAuthenticated: true,
+            details: null 
+        }
+
```
Comment (stacyp2006) [inline code/web/src/modules/user/api/state.test.js:105]: We could maybe DRY up the test code by moving `const initialState` to the top of the describe block before any it statements.

File: code/web/src/modules/user/tests/Profile.test.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        let props={
+            user: {
+                details: {
+                    name: 'Bob Forehead',
+                    email: 'Khaleta@thisisfake.com'
+                }
+            }
+        }
+
+        render(<Provider store={ store }><Profile user={props.user} logout={mockLogout} /></Provider>)
```
Comment (nathanielmillard) [inline code/web/src/modules/user/tests/Profile.test.js:25]: I have tossed a line out to see if we can get some clarity on unit testing redux components. And I am going to see what other resources we can find...

File: code/web/src/modules/user/api/state.test.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        )
+
+    })
+
+    it('should show status of logout', () => {
+        const action = {
+            type: LOGOUT
+        }
+
+        const initialState = {
```
Comment (nathanielmillard) [inline code/web/src/modules/user/api/state.test.js:99]: Glad to see this function is actually testing something.

**Review by nathanielmillard — APPROVED** [review]
I am proud of the progress we made here, and it's looking good :)

**Review by stacyp2006 — APPROVED** [review]

File: code/web/src/modules/user/api/state.test.js (hunk 1 above)
Comment (aemcdonald) [reply code/web/src/modules/user/api/state.test.js:105]: Great idea, Stacy.

**Review by aemcdonald — APPROVED** [review]
We will have to update the user reducer tests once backend has updated the user properties in the database.

**Pull request merged**